clients/unshaded/src/main/java/tachyon/client/options/OutStreamOptions.java

```diff
@@ -0,0 +1,147 @@
+    private long mTTL;
+    private UnderStorageType mUnderStorageType;
+
+    public Builder(TachyonConf conf) {
+      mBlockSize = conf.getBytes(Constants.USER_DEFAULT_BLOCK_SIZE_BYTE);
+      mHostname = null;
+      mTachyonStorageType =
+          conf.getEnum(Constants.USER_DEFAULT_TACHYON_STORAGE_TYPE, TachyonStorageType.class);
+      mUnderStorageType =
+          conf.getEnum(Constants.USER_DEFAULT_UNDER_STORAGE_TYPE, UnderStorageType.class);
+    }
+
+    /**
+     * @param blockSize the block size to use
+     * @return the builder
+     */
+    public Builder setBlockSize(long blockSize) {
+      mBlockSize = blockSize;
+      return this;
+    }
+
+    /**
+     * @param hostname the hostname to use
+     * @return the builder
+     */
+    public Builder setHostname(String hostname) {
+      mHostname = hostname;
+      return this;
+    }
+
+    /**
+     * @param tachyonStorageType the Tachyon storage type to use
+     * @return the builder
+     */
+    public Builder setTachyonStorageType(TachyonStorageType tachyonStorageType) {
+      mTachyonStorageType = tachyonStorageType;
+      return this;
+    }
+
+    /**
+     * @param underStorageType the under storage type to use
+     * @return the builder
+     */
+    public Builder setUnderStorageType(UnderStorageType underStorageType) {
+      mUnderStorageType = underStorageType;
+      return this;
+    }
+
+    public Builder setTTL(long ttl) {
+      mTTL = ttl;
+      return this;
+    }
+
+    /**
+     * Builds a new instance of <code>ClientOptions</code>
+     *
+     * @return a <code>ClientOptions</code> instance
+     */
+    public OutStreamOptions build() {
+      return new OutStreamOptions(this);
+    }
+  }
+
+  private final long mBlockSize;
+  private final String mHostname;
+  private final TachyonStorageType mTachyonStorageType;
+  private final UnderStorageType mUnderStorageType;
+  private final long mTTL;
+
+  /**
+   * @return the default <code>ClientOptions</code>
+   */
+  public static OutStreamOptions defaults() {
+    return new Builder(new TachyonConf()).build();
+  }
+
+  private OutStreamOptions(OutStreamOptions.Builder builder) {
+    mBlockSize = builder.mBlockSize;
+    mHostname = builder.mHostname;
+    mTachyonStorageType = builder.mTachyonStorageType;
+    mTTL = builder.mTTL;
+    mUnderStorageType = builder.mUnderStorageType;
+  }
+
+  /**
+   * @return the block size
+   */
+  public long getBlockSize() {
+    return mBlockSize;
+  }
+
+  /**
+   * @return the hostname
+   */
+  public String getHostname() {
+    return mHostname;
+  }
+
+  /**
+   * @return the cache type
+   */
+  public TachyonStorageType getTachyonStorageType() {
+    return mTachyonStorageType;
+  }
+
+  public long getTTL() { return mTTL; }
+
+  /**
+   * @return the under storage type
+   */
+  public UnderStorageType getUnderStorageType() {
+    return mUnderStorageType;
+  }
+
+
+  public InStreamOptions toInStreamOptions() {
```

Comment:
Good point, I removed it and instead added the logic to the intergration tests that need it.